packages/otp2-tile-overlay/src/index.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  ConfiguredCompany,
+  MapLocationActionArg,
+} from "@opentripplanner/types"
+import { Layer, Popup,  Source, useMap } from "react-map-gl"
+import EntityPopup from "@opentripplanner/map-popup"
+import React, { useEffect, useState } from "react"
+// eslint-disable-next-line prettier/prettier
+import type { EventData } from "mapbox-gl"
+import { LAYER_PAINT } from "./util"
+
+const OTP2TileLayerWithPopup = ({
+  configCompanies,
+  id,
+  network,
+  setLocation,
+  setViewedStop,
+  onMapClick,
+  type
+}: {
+  /**
+   * Optional configuration item which allows for customizing properties of scooter and
+   * bikeshare companies. If this is provided, scooter/bikeshare company names can be rendered in the
+   * default scooter/bike popup.
+   */
+  configCompanies?: ConfiguredCompany[]
+  id: string
+  name?: string
+  /**
+   * If `network` is specified, the layer will be filtered to only show vehicles from
+   * that network
+   */
+  network?: string
+  /**
+   * A method fired when a stop is selected as from or to in the default popup. If this method
+   * is not passed, the from/to buttons will not be shown.
+   */
+  setLocation?: (location: MapLocationActionArg) => void
+  /**
+   * A method fired when the stop viewer is opened in the default popup. If this method is
+   * not passed, the stop viewer link will not be shown.
+   */
+  setViewedStop?: ({ stopId }: { stopId: string }) => void
+  /**
+   * An optional method to override the map click handler. If a method is passed, NO POPUPS
+   * WILL APPEAR ON CLICK. The implementer will be responsible for handling all click events
+   * in accordance with the MapLibreGL api.
+   */
+  onMapClick?: (event: EventData) => void
+  /**
+   * Determines which layer of the OTP2 tile data to display. Also determines icon color.
+   */
+  type: string
```

Comment:
Should this a list of permitted values instead, i.e. all the known OTP layer types?

Reply:
No, as these are fully configurable in the OTP config